playwright/fixtures/index.ts

```diff
@@ -35,113 +44,242 @@ export type Fixtures = {
+      await use({ name: sliceName });
+    },
+
+    /**
+     * Page
+     */
+    page: async ({ browser }, use) => {
+      // Onboard user in Local Storage by default
+      let context = await browser.newContext({
+        storageState: {
+          cookies: [],
+          origins: [
+            {
+              origin: config.use.baseURL,
+              localStorage: [
+                {
+                  name: "persist:root",
+                  value: JSON.stringify({
+                    userContext: {
+                      userReview: {
+                        onboarding: true,
+                        advancedRepository: true,
+                      },
+                      updatesViewed: {
+                        latest: null,
+                        latestNonBreaking: null,
+                      },
+                      hasSeenChangesToolTip: true,
+                      hasSeenSimulatorToolTip: true,
+                      hasSeenTutorialsToolTip: true,
+                      authStatus: "unknown",
+                      lastSyncChange: null,
                     },
-                    updatesViewed: { latest: null, latestNonBreaking: null },
-                    hasSeenChangesToolTip: true,
-                    hasSeenSimulatorToolTip: true,
-                    hasSeenTutorialsToolTip: true,
-                    authStatus: "unknown",
-                    lastSyncChange: null,
-                  },
-                }),
-              },
+                  }),
+                },
+                {
+                  name: "slice-machine_isInAppGuideOpen",
+                  value: "false",
+                },
+              ],
+            },
+          ],
+        },
+      });
+
+      // Prevent user to be onboarded if needed
+      if (!onboarded) {
+        context = await browser.newContext({
+          storageState: {
+            cookies: [],
+            origins: [
               {
-                name: "slice-machine_isInAppGuideOpen",
-                value: "false",
+                origin: config.use.baseURL,
+                localStorage: [],
               },
             ],
           },
-        ],
-      },
-    });
+        });
+      }
+
+      // Create new page object with new context
+      const page = await context.newPage();
+
+      // Logout user by default
+      try {
+        await fs.rm(path.join(os.homedir(), ".prismic"));
+      } catch (error) {
+        // Ignore since it means the user is already logged out
+      }
+
+      // Login user if needed
+      if (loggedIn) {
+        // In CI we define a PRISMIC_URL env variable to fasten the tests
+        let prismicUrl: string | undefined = process.env["PRISMIC_URL"];
+
+        // In local we get the Prismic URL from the browser, it helps to avoid
+        // switching manually between Wroom and Prismic
+        if (!prismicUrl) {
+          await page.route(
+            "*/**/_manager",
```

Comment:
`@slicemachine/manager/test` provides with a `createSliceMachineManagerMSWHandler()` test helper to mock manager RPC requests.

You can see it being used in `slice-machine-ui` here: https://github.com/prismicio/slice-machine/blob/d061d339995c8d849ccf8d1f7f84a22e7dffadd1/packages/slice-machine/test/pages/slices.test.tsx#L56-L61

Is there a way we can make a Playwright version of that helper? The helper is specific to MSW, but I think we could adapt it for Playwright's network APIs.

Reply:
I will take a look, but the interception should be made on the browser side, so I'm not sure how it will work. I will take a look and see what we can do, good idea!

Reply:
Discussed together 🙂